engine/src/main/java/org/hibernate/search/v6poc/engine/impl/SearchMappingRepositoryBuilderImpl.java

```diff
@@ -279,6 +269,40 @@ void createAndAddMapper(Map<MappingKey<?>, Mapper<?>> mappers,
 			}
 		}
 
+		void createAndAddMapping(Map<MappingKey<?>, MappingImplementor<?>> mappings) {
+			if ( mappings.containsKey( mappingKey ) ) {
+				throw new AssertionFailure(
+						"Found two mapping initiators using the same key."
+						+ " There is a bug in the mapper, please report it."
+				);
+			}
+
+			try {
+				MappingImplementor<M> mapping = mapper.build();
+				mappings.put( mappingKey, mapping );
+			}
+			catch (MappingAbortedException e) {
+				ContextualFailureCollector failureCollector = buildContext.getFailureCollector();
+				Throwable cause = e.getCause();
+				if ( cause != null ) {
+					failureCollector.add( cause );
+				}
+				Throwable[] suppressed = e.getSuppressed();
+				for ( Throwable throwable : suppressed ) {
+					failureCollector.add( throwable );
+				}
+
+				if ( !failureCollector.hasFailure() ) {
```

Comment:
I think I would have put this test at the beginning of the catch because we expect the collector to have failures from the get go, before adding the potential cause and suppressed exceptions.

Reply:
Right, done.